delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -221,17 +221,47 @@ public TableAssessmentResponse assessTable(DraftId draftId, Configurer configure
     Draft draft = getDraft(draftId);
     DeltaConfig deltaConfig = draft.getConfig();
     deltaConfig = evaluateMacros(draftId.getNamespace(), deltaConfig);
-    Stage target = deltaConfig.getTarget();
-    if (target == null) {
-      throw new InvalidDraftException("Cannot assess a table without a configured target.");
-    }
     SourceTable selectedTable = deltaConfig.getTables().stream().filter(
       streamTable -> db.equals(streamTable.getDatabase()) && table.equals(streamTable.getTable()) &&
         (schema == streamTable.getSchema() || schema != null && schema.equals(streamTable.getSchema()))).findAny()
       .orElseThrow(() -> new IllegalArgumentException(String
         .format("Table '%s' in database '%s' and schema '%s' is not a selected table in the draft", table, db,
           schema)));
 
+    return assessTable(draftId.getNamespace(), deltaConfig, configurer, selectedTable);
+  }
+
+  /**
+   * Assess the given table detail using the plugins from the given Delta Config. Plugins will be
+   * instantiated in order to generate this assessment. This is an expensive operation.
+   *
+   * @param namespace namespace to look for macros
+   * @param deltaConfig : delta config object to assess
+   * @param configurer configurer used to instantiate plugins
+   * @return list of tables readable by the source in the draft
+   * @throws IOException if the table detail could not be read
+   * @throws Exception if there was an error creating the table registry
+   * @throws IllegalArgumentException if the number of tables is not equal to one.
+   */
+  public TableAssessmentResponse assessTable(Namespace namespace, DeltaConfig deltaConfig, Configurer configurer)
+    throws Exception {
+
+    deltaConfig = evaluateMacros(namespace, deltaConfig);
+    if (deltaConfig.getTables().size() != 1) {
```

Comment:
hmm.... this is a little bit weird. can we add a AssessTableRequest which contains a DeltaConfig and a DBTable , which makes the interface more clear and consistent with the non-on-the-fly interface ?